File: isoverlapping.m
function y=isoverlapping(stroke1,stroke2)
assert(isa(stroke1,'double'));
assert(isa(stroke2,'double'));
minx1=min(stroke1(:,1));
maxx1=max(stroke1(:,1));
minx2=min(stroke2(:,1));
maxx2=max(stroke2(:,1));
if(minx1<minx2)
    
        a=maxx1-minx2;
        b=maxx1-minx1;
        c=a/b;
        
else
    
        a=maxx2-minx1;
        b=maxx2-minx2;
        c=a/b;
        
end
if c>0        %there should be atleast 20%overlap--threshold
    y=1;
else
    y=0;
end
end

    

File: codegen/mex/isoverlapping/isoverlapping.c
/*
 * isoverlapping.c
 *
 * Code generation for function 'isoverlapping'
 *
 */

/* Include files */
#include "rt_nonfinite.h"
#include "isoverlapping.h"

/* Function Definitions */
real_T isoverlapping(real_T stroke1, real_T stroke2)
{
  real_T y;
  real_T c;
  if (stroke1 < stroke2) {
    c = (stroke1 - stroke2) / (stroke1 - stroke1);
  } else {
    c = (stroke2 - stroke1) / (stroke2 - stroke2);
  }

  if (c > 0.0) {
    /* there should be atleast 20%overlap--threshold */
    y = 1.0;
  } else {
    y = 0.0;
  }

  return y;
}

/* End of code generation (isoverlapping.c) */
